roles/pulp_webserver/tasks/generate_tls_certificates.yml

```diff
@@ -0,0 +1,69 @@
+---
+- name: Set python-cryptography package name
+  set_fact:
+    python_cryptography: python-cryptography
+  when: ansible_python.version.major == 2
+
+- name: Set python-cryptography package name
+  set_fact:
+    python_cryptography: python3-cryptography
+  when: ansible_python.version.major == 3
+
+- name: Ensure python-cryptography is installed
+  become: true
+  package:
+    name: '{{ python_cryptography }}'
+
+- name: Generate CA key
+  openssl_privatekey:
+    path: '{{ pulp_webserver_tls_folder }}/root.key'
+
+- name: Generate CA CSR
+  openssl_csr:
+    path: '{{ pulp_webserver_tls_folder }}/root.csr'
+    privatekey_path: '{{ pulp_webserver_tls_folder }}/root.key'
+    common_name: '{{ ansible_fqdn }}'
+    organization_name: Pulp
+    country_name: US
+    basic_constraints: 'CA:TRUE'
+
+- name: Generate CA certificate
+  openssl_certificate:
+    path: '{{ pulp_webserver_tls_folder }}/root.crt'
+    csr_path: '{{ pulp_webserver_tls_folder }}/root.csr'
+    privatekey_path: '{{ pulp_webserver_tls_folder }}/root.key'
+    provider: selfsigned
+
+- name: Generate private keys
+  openssl_privatekey:
+    path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.key'
+
+- name: Generate CSRs standalone
+  openssl_csr:
+    path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.csr'
+    privatekey_path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.key'
+    common_name: '{{ ansible_fqdn }}'
+    subject_alt_name: 'DNS:{{ ansible_fqdn }}'
+    key_usage:
+      - keyEncipherment
+      - dataEncipherment
+    extended_key_usage:
+      - serverAuth
+
+- name: Generate certificates
+  openssl_certificate:
+    path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.crt'
+    csr_path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.csr'
+    privatekey_path: '{{ pulp_webserver_tls_folder }}/pulp_webserver.key'
+    provider: ownca
+    ownca_path: '{{ pulp_webserver_tls_folder }}/root.crt'
+    ownca_privatekey_path: '{{ pulp_webserver_tls_folder }}/root.key'
+    ownca_not_after: '+824d'
+
+- name: Cleanup CSR files
+  file:
+    path: '{{ pulp_webserver_tls_folder }}/{{ item }}'
+    state: absent
+  loop:
+    - root.csr
+    - pulp_webserver.csr
```

Comment:
This will fail idempotency with the task that generates them above.

Reply:
Might be addressed by https://github.com/pulp/pulp_installer/pull/325#discussion_r448825714 based on the option chosen